### ISSUE TITLE
New order notification not being sent to buyers

### ISSUE THREAD

**Issue by SukcesStrony** [woocommerce/woocommerce#12044]
## EXPLANATION OF THE ISSUE

The issue described below confuses buyers, makes it harder for store owners to notice and manage unpaid orders and - most of all - infringes the European Union law "The Directive on Consumer Rights", available at http://ec.europa.eu/justice/consumer-marketing/files/crd_guidance_en.pdf - further called "EU Directive".

There is a "New order" notification, but it only gets sent when the order is paid for. The notification about a new order isn't sent if the payment is not made (for example the payment gateway failed or the user didn't have funds on his credit card). Fixes to be applied:

1. The "New order" e-mail should be **sent in all cases right after clicking the "Confirm purchase" button**. Clicking this button is a confirmation of a binding contract, according to the EU Directive 5.2.2. Requirements for the order confirmation button (page 32). Even if the payment is not made, the user should know that he signed an agreement and should receive an e-mail confirmation, which leads us to:
2. The "New order" e-mail should be received by **both the admin and the buyer** (currently it's received only by the admin). According to the EU Directive 5.4. Confirmation of the contract (page 35) the buyer should receive this confirmation right after clicking the purchase button, no matter if he paid or not. This leads us to:
3. The "New order" **e-mail should contain an explanation on how to pay for the order**. For example, it should start with: _"This is a confirmation of placing your order. If you paid for it, you should get a separate e-mail stating, that the order is "In progress". If you didn't pay for the order, didn't receive that e-mail or aren't sure if the payment went through, please go to the order page to check the status and make the payment if necessary."_. The "order page" text should be a link to the order, which by default lets the users pay or change the payment method.
4. **Added by editing:** The stock should also be reduced right after clicking the purchase button to secure the goods of the contract. Not sure if it works this way right now or is the stock reduced only when the payment is made. If the latter, this would also introduce the problem of 2 people ordering the same thing which has a stock of 1 and paying with some delay.

## WHY IS THIS IMPORTANT

1. It infringes the EU Directive, so that every store owner in the EU is basically held responsible.
2. If an order is placed but not paid for (for example the gateway failed), the user doesn't know how to recover and the admin doesn't even know that an order was placed and the user had trouble with paying for it. An e-mail sent right after clicking the purchase button would let the user and admin know that an agreement has been made and how to check and pay for the order if the user hasn't done so yet.

**Comment by mikejolley** [comment]
I don't feel this is entirely accurate or desirable. It will ultimately lead to spam orders + emails and much more undesirable stuff in the database. Orders right now are not stored in the DB in the majority of cases unless payment will be successful, at which point the order is stored and admin/user are both notified. 

Quick points, 

1. We don't have a "Confirm purchase" button. We use "place order" by default. If you've changed the wording and thus made yourself subject to this clause thats not really a WC problem
2. Regarding #2, this isn't true. Both admin and customer get emails at the same times and for the same order status changes.
3. We don't do this except for invoices which are sent via the backend. A regular order is not an invoice.
4. Stock is held if orders are pending payment, but not until the order is created. If you use a CC gateway which takes payment during checkout, there is no need to do a reduction unless that payment is successful.

To note, I am in the UK (still in the EU for now :p) and when I make an order but enter wrong payment details for example, I never get an email stating this and _I am free to just not pay if I so choose._ 

If you go on amazon and checkout, but change mind or card payment fails, you don't get an email either afaik. You can try again or leave. Your choice. 

Ultimately this can be controlled by your gateway anyway if you insist on having it work in this way. You'll just want the gateway to mark paid orders processing, and failed/unpaid/incomplete orders on-hold. That will trigger emails are storage etc. If you've changed button wording you'll probably want to do this instead. Customise your payment gateway.

Hope that makes sense. Feel free to reply but I'm closing this for now.

**Comment by SukcesStrony** [comment]
> It will ultimately lead to spam orders + emails and much more undesirable stuff in the database. 

Orders right now are not stored in the DB in the majority of cases unless payment will be successful
Currently orders placed but unpaid are still stored as "awaiting payment" and then "failed" (if not paid on time) orders. I imagine there's a bit more stuff with a paid order, but the database isn't going to clog itself due to this. If spam orders were a problem, it would be apparent even if the current state - they would still be in the database as said in the 1st sentence. The email notifications can be turned off. Finally, even if spam becomes a problem, there are ways to fix it and it shouldn't block us from developing.

> We don't have a "Confirm purchase" button. We use "place order" by default. If you've changed the wording and thus made yourself subject to this clause that's not really a WC problem

I wasn't referring to the text of the button as a problem. It's very easy to change and I just wasn't sure what's it called originally. This is not an issue though, as it's easily changeable. The EU countries in particular take care of that usually on the translation level.

> Regarding #2, this isn't true. Both admin and customer get emails at the same times and for the same order status changes.

You are correct and I was mistaken on this point. The admin get's the "New Order" e-mail after the payment was made, and the user gets the "Processing order" e-mail after the payment was made. 

> To note, I am in the UK (still in the EU for now :p) and when I make an order but enter wrong payment details for example, I never get an email stating this and I am free to just not pay if I so choose.

This is not true. All EU member states were to implement the Directive until 2014. The UK also did that under "The Consumer Contracts (Information, Cancellation and Additional Charges) Regulations 2013". The wording had to be almost exact and it is. Here are references to the points linked above for your country: http://www.legislation.gov.uk/uksi/2013/3134/regulation/14/made and http://www.legislation.gov.uk/uksi/2013/3134/regulation/16/made

**As the EU is the largest economy in the world (the US is close though), this should be a priority.**

> If you go on amazon and checkout, but change mind or card payment fails, you don't get an email either afaik. 

I'm not sure if they do it like that, but that's Amazon - one of the largest companies in the world, which can have exempts. They are also registered in the US and might have a way around that law. Either way, it doesn't matter what they do, what matters is what law is binding to us.

> Ultimately this can be controlled by your gateway anyway if you insist on having it work in this way.

This would require custom coding for each payment gateway in each country in the EU. It is not possible. Setting unpaid orders to "on hold" would cause a mess - the user and admin would be wondering why the order is on hold, was the payment made, how to pay if it wasn't etc... "On hold" is also not a confirmation of a binding contract.

As I said, **this isn't only about the EU regulations. It's mostly about buyers. Let me give you an example**. Let's say you added 50 products to your cart, which took 2 hours. The payment gateway asks for your CC info and then a "404 not found" error occurs on the next page of the payment gateway. You have no idea what to do next. Did the payment go through? Should you shop again for 2 hours to fill the cart again, which at the end might fail again? Usually such users just leave the site, while it would be so easy to have a "pay if you haven't already" link in the order placement email. On the admin side, he could contact the buyer and ask what went wrong and if he can help complete the purchase. This makes a lot of sense for both the buyer and the admin, law regulations aside.

To sum up, here are the requests which I find still very valid:

1. The "New order" e-mail should be sent in all cases right after clicking the "Confirm purchase" button to both the buyer and the admin (which can be toggled on/off).
2. The "New order" e-mail should contain an explanation on how to pay for the order (explained in the first post).

I welcome your reply as this is a seriously taken matter in the Polish legal system and cash penalties are applied here for such things.

**Comment by mikejolley** [comment]
> Orders right now are not stored in the DB in the majority of cases unless payment will be successful
Currently orders placed but unpaid are still stored as "awaiting payment" and then "failed" (if not paid on time) orders. I imagine there's a bit more stuff with a paid order, but the database isn't going to clog itself due to this. If spam orders were a problem, it would be apparent even if the current state - they would still be in the database as said in the 1st sentence. The email notifications can be turned off. Finally, even if spam becomes a problem, there are ways to fix it and it shouldn't block us from developing.

The pending/failed orders are for PayPal. Other gateways, such as Stripe, don't create an order if the payment fails due to user error.

Your 404 example is valid, however, if that did occur wouldn't it be due to server/site error? If that happened, and execution stopped, even order creation + email could fail.

> The "New order" e-mail should be sent in all cases right after clicking the "Confirm purchase" button to both the buyer and the admin (which can be toggled on/off).
> The "New order" e-mail should contain an explanation on how to pay for the order (explained in the first post).

Would a 'failed' email, with a pay link, satisfy this? Only on payment error. The user would still be on checkout however, thats my concern. And you'd have that order somewhere.

>  If the contract places the consumer under an obligation to pay, the trader must make the consumer aware in a clear and prominent manner, and directly before the consumer places the order, of the information listed in paragraphs (a), (f), (g), (h), (s) and (t) of Schedule 2.

^ This is from your link. Clicking that place order button doesn't necessarily mean they are under obligation to pay. For an order through paypal example, you could change your mind before payment, no? That doc also states you need to make the button state "order with obligation to pay". We don't do that in core either.

**Comment by mikejolley** [comment]
@SukcesStrony Actually thats another point. If your payment does fail, your cart is not cleared, so the case you mentioned about the user not being able to pay again is not entirely true. They'd just re-checkout.

What gateway are you using? I wonder if that is just faulty?
